tests/gnu/testlet/TestUtils.java

```diff
@@ -0,0 +1,91 @@
+package gnu.testlet;
+
+import java.io.*;
+
+public class TestUtils {
+    public static byte[] read(InputStream is) throws IOException {
+        int l = is.available();
+        byte[] buffer = new byte[l+1];
+        int length = 0;
+
+        while ((l = is.read(buffer, length, buffer.length - length)) != -1) {
+            length += l;
+            if (length == buffer.length) {
+                byte[] b = new byte[buffer.length + 4096];
+                System.arraycopy(buffer, 0, b, 0, length);
+                buffer = b;
+            }
+        }
+
```

Comment:
There is a bug with `read()` that we doesn't remove the unused bytes if the buffer is large that needed. It could be fixed by add something like this:
```Java
if (length < buffer.length) {
  byte[] b = new byte[length];
  System.arraycopy(buffer, 0, b, 0, length);
  buffer = b;
}
```